planning/behavior_velocity_planner/autoware_behavior_velocity_blind_spot_module/src/util.cpp

```diff
@@ -681,4 +684,57 @@ std::optional<StopPoints> generate_stop_points(
     std::nullopt, stop_points_list.instant_stopline, stop_points_list.critical_stopline};
 }
 
+tl::expected<double, std::string> calc_ego_to_blind_spot_lanelet_lateral_gap(
+  const autoware_utils::LinearRing2d & ego_footprint,
+  const lanelet::ConstLanelets & last_lanelets_before_turning,
+  const autoware::experimental::lanelet2_utils::TurnDirection & turn_direction)
+{
+  if (ego_footprint.size() != 7) {
+    return tl::make_unexpected("Unexpected ego_footprint's size.");
+  }
+
+  // left front is 6, left back is 4, right_front is 1, right_back is 3
+  const auto front_idx = (turn_direction == TurnDirection::Left) ? 6 : 1;
+  const auto back_idx = (turn_direction == TurnDirection::Left) ? 4 : 3;
+  const auto ego_side =
+    autoware_utils::Segment2d{ego_footprint[front_idx], ego_footprint[back_idx]};
+
+  std::vector<autoware_utils::Point2d> line;
+  for (const auto & ll : last_lanelets_before_turning) {
+    const auto & attention_area_road_boundary = lanelet::utils::to2D(
+      (turn_direction == TurnDirection::Left) ? ll.leftBound() : ll.rightBound());
+    const auto ll_2d = lanelet::utils::to2D(attention_area_road_boundary);
+
+    for (const auto & ls : ll_2d) {
+      line.emplace_back(ls.x(), ls.y());
+    }
+  }
+
+  if (line.size() < 2) {
+    return tl::make_unexpected("Not enough points in lanelet boundary.");
+  }
+
+  std::vector<autoware_utils::Segment2d> segments;
+  segments.reserve(line.size() - 1);
+  for (size_t i = 0; i + 1 < line.size(); ++i) {
```

Comment:
```suggestion
  for ([point1, point2) = ranges::view::zip(line, line | ranges::views::drop(1)) {
```

Reply:
done in https://github.com/autowarefoundation/autoware_universe/pull/11265/commits/894fcf90ae086af6c888b1b597b064023a7a70b1